### PR TITLE
update libraries to fix vulnerabilities

### PR DESCRIPTION
**Pull request by dhiaayachi**
upgrade few dependencies to improve security stature.

### CODE REVIEW

**Comment by dhiaayachi** [comment]
Azure tests are failing but I think it's related to the azure test environment being broken

**Review by zalimeni — APPROVED** [review]
Thank you!

**Comment by dhiaayachi** [comment]
@zalimeni You can safely ignore deleted files as those are govendor files. I replaced it with go.mod as govendor is EOL and I needed to upgrade dependencies in that package.

File: provider/gce/gce_discover_test.go [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package gce_test
 
 import (
+	"github.com/hashicorp/go-discover"
 	"io/ioutil"
 	"log"
 	"os"
 	"testing"
 
-	discover "github.com/hashicorp/go-discover"
```
Comment (zalimeni) [inline provider/gce/gce_discover_test.go:9]: IIRC `golangci-lint` likes to complain about undefined types for hyphenated aliases. Not sure whether it's complaining here but thought worth calling out.

Also super minor but I think formatting should place that dep w/ other `github.com` as before. NBD and not a blocker.

**Review by zalimeni — APPROVED** [review]

File: provider/gce/gce_discover_test.go (hunk 1 above)
Comment (dhiaayachi) [reply provider/gce/gce_discover_test.go]: reverted it, just in case 😅

**Pull request merged**